### ISSUE TITLE
Cannot change column 'billable_ptr_id': used in a foreign key constraint 'juntagrico_subscript_subscription_id_1f724c85_fk_juntagric'

### ISSUE THREAD

**Issue by bkrumnow** [juntagrico/juntagrico#423]
Hey,

I am trying to run Juntagrico 1.4.1 with MariaDB. I had previously installed 1.2.3. Then tried migrating to 1.4.1 running into a foreign key constraint error (see code below). I tried similar things with SQLite, which worked fine. I also tried installing Juntagrico with a clean database with MariaDB leading to the same issue. Any suggestions?

```
Operations to perform:
  Apply all migrations: admin, auth, contenttypes, defender, impersonate, juntagrico, sessions, sites
Running migrations:
  Applying juntagrico.0034_auto_20210415_2041...Traceback (most recent call last):
  File "/srv/juntagrico/venv/lib/python3.7/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/srv/juntagrico/venv/lib/python3.7/site-packages/django/db/backends/mysql/base.py", line 73, in execute
    return self.cursor.execute(query, args)
  File "/srv/juntagrico/venv/lib/python3.7/site-packages/MySQLdb/cursors.py", line 206, in execute
    res = self._query(query)
  File "/srv/juntagrico/venv/lib/python3.7/site-packages/MySQLdb/cursors.py", line 319, in _query
    db.query(q)
  File "/srv/juntagrico/venv/lib/python3.7/site-packages/MySQLdb/connections.py", line 259, in query
    _mysql.connection.query(self, query)
MySQLdb._exceptions.OperationalError: (1833, "Cannot change column 'billable_ptr_id': used in a foreign key constraint 'juntagrico_subscript_subscription_id_1f724c85_fk_juntagric' of table 'junta_db.juntagrico_subscriptionmembership'")

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "manage.py", line 21, in <module>
    main()
  File "manage.py", line 17, in main
    execute_from_command_line(sys.argv)
  File "/srv/juntagrico/venv/lib/python3.7/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "/srv/juntagrico/venv/lib/python3.7/site-packages/django/core/management/__init__.py", line 413, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/srv/juntagrico/venv/lib/python3.7/site-packages/django/core/management/base.py", line 354, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/srv/juntagrico/venv/lib/python3.7/site-packages/django/core/management/base.py", line 398, in execute
    output = self.handle(*args, **options)
  File "/srv/juntagrico/venv/lib/python3.7/site-packages/django/core/management/base.py", line 89, in wrapped
    res = handle_func(*args, **kwargs)
  File "/srv/juntagrico/venv/lib/python3.7/site-packages/django/core/management/commands/migrate.py", line 246, in handle
    fake_initial=fake_initial,
  File "/srv/juntagrico/venv/lib/python3.7/site-packages/django/db/migrations/executor.py", line 117, in migrate
    state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
  File "/srv/juntagrico/venv/lib/python3.7/site-packages/django/db/migrations/executor.py", line 147, in _migrate_all_forwards
    state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
  File "/srv/juntagrico/venv/lib/python3.7/site-packages/django/db/migrations/executor.py", line 227, in apply_migration
    state = migration.apply(state, schema_editor)
  File "/srv/juntagrico/venv/lib/python3.7/site-packages/django/db/migrations/migration.py", line 126, in apply
    operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
  File "/srv/juntagrico/venv/lib/python3.7/site-packages/django/db/migrations/operations/fields.py", line 244, in database_forwards
    schema_editor.alter_field(from_model, from_field, to_field)
  File "/srv/juntagrico/venv/lib/python3.7/site-packages/django/db/backends/base/schema.py", line 595, in alter_field
    old_db_params, new_db_params, strict)
  File "/srv/juntagrico/venv/lib/python3.7/site-packages/django/db/backends/base/schema.py", line 821, in _alter_field
    fragment[1],
  File "/srv/juntagrico/venv/lib/python3.7/site-packages/django/db/backends/base/schema.py", line 145, in execute
    cursor.execute(sql, params)
  File "/srv/juntagrico/venv/lib/python3.7/site-packages/django/db/backends/utils.py", line 66, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/srv/juntagrico/venv/lib/python3.7/site-packages/django/db/backends/utils.py", line 75, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/srv/juntagrico/venv/lib/python3.7/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/srv/juntagrico/venv/lib/python3.7/site-packages/django/db/utils.py", line 90, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/srv/juntagrico/venv/lib/python3.7/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/srv/juntagrico/venv/lib/python3.7/site-packages/django/db/backends/mysql/base.py", line 73, in execute
    return self.cursor.execute(query, args)
  File "/srv/juntagrico/venv/lib/python3.7/site-packages/MySQLdb/cursors.py", line 206, in execute
    res = self._query(query)
  File "/srv/juntagrico/venv/lib/python3.7/site-packages/MySQLdb/cursors.py", line 319, in _query
    db.query(q)
  File "/srv/juntagrico/venv/lib/python3.7/site-packages/MySQLdb/connections.py", line 259, in query
    _mysql.connection.query(self, query)
django.db.utils.OperationalError: (1833, "Cannot change column 'billable_ptr_id': used in a foreign key constraint 'juntagrico_subscript_subscription_id_1f724c85_fk_juntagric' of table 'junta_db.juntagrico_subscriptionmembership'")
```

**Comment by motu55** [comment]
this seems to be a problem with your db settings maybe disabling the foreign key check for some amount of time.
since we inhouse use postgresdb we cannot really help here

**Comment by pascalfree** [comment]
With version 1.6 of Juntagrico a fresh install with MariaDB or MySQL will successfully migrate, because the migrations have been squashed and the table will be created with the right foreign key field type from the start.

MySQL is now tested in the CI process and will receive better support.
One known issue, will be fixed with #699.

If you still have an existing installation, that needs to be migrated, please try if Juntagrico 1.6 fixes the issue (or rather django 4.2 that comes with it).
